Source/Tools/FEXServer/CoreDumpService.h

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: MIT
+#pragma once
+#include "Common/FEXServerClient.h"
+#include "Unwind/FileMapping.h"
+#include "SocketConnectionHandler.h"
+
+#include <FEXCore/fextl/map.h>
+#include <FEXCore/fextl/list.h>
+#include <FEXCore/fextl/string.h>
+#include <FEXCore/fextl/unordered_map.h>
+#include <FEXCore/fextl/vector.h>
+
+#include <cstring>
+#include <linux/limits.h>
+#include <poll.h>
+#include <sys/socket.h>
+#include <thread>
+
+namespace Unwind {
+class Unwinder;
+}
+
+namespace CoreDumpService {
+  class CoreDumpClass final : public FEXServer::SocketConnectionHandler {
+    public:
+      void OnShutdown() override;
+      SocketConnectionHandler::FDEventResult HandleFDEvent(struct pollfd &Event) override;
+
+      struct Description {
+        uint32_t pid;
+        uint32_t tid;
+        uint32_t uid;
+        uint32_t gid;
+        uint32_t Signal;
+        uint64_t Timestamp;
+        uint8_t HostArch;
+        uint8_t GuestArch;
+      };
+
+      // Initializes this CoreDumpClass's execution thread and associated sockets.
+      // Returns one socket of the socketpair for the paired connection.
+      int InitExecutionThread() {
+        int SVs[2];
+        int Result = socketpair(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0, SVs);
+        if (Result == -1) {
+          return -1;
+        }
+
+        PollFDs.push_back(pollfd {
+          .fd = SVs[0],
+          .events = POLLIN | POLLHUP | POLLERR | POLLNVAL | POLLREMOVE | POLLRDHUP,
+          .revents = 0,
+        });
+        ExecutionThread = std::thread(&CoreDumpClass::ExecutionFunc, this);
+
+        ServerSocket = SVs[0];
```

Comment:
This looks like it matters only academically, but ServerSocket should be initialized before starting the `ExecutionThread`, shouldn't it?